src/expressions.md

```diff
@@ -214,43 +217,143 @@ assert_eq!(5, x);
+1. If it's possible, Rust will then repeat steps 1-5 with
+  `<A as std::ops::Deref>::Target`, and insert a dereference operator.
+
+Note: in steps 1-4 the method doesn't have to be implemented on `A`. For
+example in step 3 the trait could be implemented for `&A` with the method
+taking `self`. If a step is reached where there is more than one possible
+method, then it is a compiler error. To resolve this use THIS ISN'T IN THE
+REFERENCE!!!!!!!!
+
+```rust
+let pi: Result<f32, _> = "3.14".parse();
+let log_pi = pi.unwrap_or(0.0).log(2.72);
+# assert!(1.14 < log_pi && log_pi < 1.15)
+```
 
 ## Field expressions
 
 A _field expression_ consists of an expression followed by a single dot and an
-identifier, when not immediately followed by a parenthesized expression-list
-(the latter is a [method call expression](#method-call-expressions)). A field
-expression denotes a field of a [struct](types.html#struct-types).
+[identifier](identifiers.html), when not immediately followed by a parenthesized expression-list
+(the latter is always a [method call expression](#method-call-expressions)). A
+field expression denotes a field of a [struct](types.html#struct-types). To
+call a function stored in a struct parentheses are needed around the field
+epression
 
 ```rust,ignore
 mystruct.myfield;
 foo().x;
 (Struct {a: 10, b: 20}).a;
+mystruct.method();          // Method expression
+(mystruct.function_field)() // Call expression
 ```
 
 A field access is an [lvalue](expressions.html#lvalues-rvalues-and-temporaries)
 referring to the value of that field. When the type providing the field
 inherits mutability, it can be [assigned](#assignment-expressions) to.
 
-Also, if the type of the expression to the left of the dot is a
-pointer, it is automatically dereferenced as many times as necessary
-to make the field access possible. In cases of ambiguity, we prefer
-fewer autoderefs to more.
+Also, if the type of the expression to the left of the dot is a pointer, it is
+automatically dereferenced as many times as necessary to make the field access
+possible. In cases of ambiguity, we prefer fewer autoderefs to more.
+
+## Call expressions
+
+A _call expression_ consists of an expression followed by a parenthesized
+expression-list. It invokes a function, providing zero or more input variables.
+If the function eventually returns, then the expression completes. The type of
+the expression that is called must implement one of the `std::ops::Fn`,
+`std::ops::FnMut` or `std::ops::FnOnce` traits, which differ in whether they
+take the type by reference, mutable refernece, or take ownership respectively.
+Some examples of call expressions:
+
+```rust
+# fn add(x: i32, y: i32) -> i32 { 0 }
+let three: i32 = add(1i32, 2i32);
+let name: &'static str = (|| "Rust")();
+```
+
+## Lambda expressions
+
+A _lambda expression_ (sometimes called an "anonymous function expression")
+defines a closure and denotes it as a value, in a single expression. A lambda
```

Comment:
Am I using 'lambda expression' and 'closure' correctly? Is there even a (single) correct choice?